File: class/Student.cpp
// Student.cpp
#include "Student.h"
Student::Student(string name, char gender,
                 int age, string major)
  // Initialization of inherited data
  // members is done via
  // member initialization list
  : Person(name, gender, age), major(major)
{}

string Student::getMajor() const {
  return major;
}

void Student::setMajor(string major) {
  this->major = major;
}

void Student::print() const {
  // Call print() of the base class
  Person::print();
  cout << "Major: " << major << endl;
}

File: src/Student.java
// Student.java

// extends is used for inheritance
// No access modifier shoudl be placed after extends
// since only public inheritance is available
public class Student extends Person {
  private String major;
  
  public Student(String name, char gender, int age, String major) {
    // Initialization of inherited instance variables is done 
    // using the keyword super
    super(name, gender, age);
    this.major = major;
  }
  
  public String getMajor() { return major; }  
  
  public void setMajor(String major) { this.major = major; }
  
  public void print() {
    // Call print() of the superclass
    super.print();
    System.out.println("Major: " + major);
  }
}